cpp/src/barretenberg/stdlib/recursion/verification_key/verification_key.hpp

```diff
@@ -264,23 +325,17 @@ template <typename Curve> struct verification_key {
     field_t<Composer> num_public_inputs;
     field_t<Composer> z_pow_n;
 
-    // NOTE: This does not strictly need to be a circuit type. It can be used to check in the circuit
-    // if a proof contains any aggregated state.
-    bool_t<Composer> contains_recursive_proof;
-
     evaluation_domain<Composer> domain;
 
     std::map<std::string, typename Curve::g1_ct> commitments;
 
     // Native data:
-
     std::shared_ptr<VerifierReferenceString> reference_string;
-
     PolynomialManifest polynomial_manifest;
-
+    // Used to check in the circuit if a proof contains any aggregated state.
+    bool contains_recursive_proof = false;
+    std::vector<uint32_t> recursive_proof_public_input_indices;
     size_t program_width = 4;
-
-    std::shared_ptr<plonk::verification_key> base_key;
```

Comment:
I suppose the reasoning for getting rid of this was: dsl already produced circuit types, so the options were to either: instantiate a vk from this and use existing code paths OR refactor the recursive verifier a bit. You noted that the base key could pretty easily go away, so you opted for the latter. Is that right?